### PR TITLE
feat(PDS-58): Bodies

### PR DESCRIPTION
**Pull request by RiccardoCorona-MiaPlatform**
### Description

Bodies components implementation

### Issue ticket number and link

[Componente Body (s-m-l)](https://makeitapp.atlassian.net/browse/PDS-59)

### CODE REVIEW

File: src/components/Typography/BodyX/BodyX.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  isBold,
+  size,
+}: BodyXProps): ReactElement => {
+  const bodyClassName = useMemo(() => classnames(
+    [
+      size === S && bodyS,
+      size === M && bodyM,
+      size === L && bodyL,
+    ]
+  ), [size])
```
Comment (Danielecina) [inline src/components/Typography/BodyX/BodyX.tsx:111]:
```suggestion
  const bodyClassName = useMemo(() => classnames([
    size === S && bodyS,
    size === M && bodyM,
    size === L && bodyL,
  ]), [size])
```

File: src/components/Typography/BodyX/BodyX.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  return (
+    <AntParagraph
+      className={bodyClassName}
+      copyable={copyable}
+      ellipsis={(ellipsis === true && { rows: 2, tooltip: { children } }) || ellipsis}
```
Comment (Danielecina) [inline src/components/Typography/BodyX/BodyX.tsx:117]: memoizziamolo sopra se no genera ad ogni rerender un oggetto nuovo

**Review by Danielecina — DISMISSED** [review]

**Review by Danielecina — DISMISSED** [review]

File: src/components/Typography/BodyX/BodyX.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+const bodyMEllipsisWithTooltip = { rows: 3, tooltip: bodyMLongText }
+const bodyLEllipsisWithTooltip = { rows: 3, tooltip: bodyLLongText }
+const bodySBoldEllipsisWithTooltip = { rows: 3, tooltip: bodySBoldLongText }
+const bodyMBoldEllipsisWithTooltip = { rows: 3, tooltip: bodyMBoldLongText }
+const bodyLBoldEllipsisWithTooltip = { rows: 3, tooltip: bodyLBoldLongText }
+
+const meta = {
+  title: 'Components/Typography/BodyX',
+  component: BodyX,
+  args: {
```
Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.stories.tsx:45]: What do you think about adding in the design tab the link to fontsource? (google-font is not embeddable)

```
const fontUrl = 'https://fontsource.org/fonts/inter'

...

parameter: {
       design: { type: 'iframe', name: 'Font Definition', url: fontUrl },
}
```

File: src/components/Typography/BodyX/BodyX.module.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+.bodyS {
+  font-size: calc(var(--typography-bodyS-fontSize, 14) * 1px);
```
Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.module.css:20]: we should set relative line heights

File: src/components/Typography/BodyX/BodyX.module.css [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  line-height: calc(var(--typography-bodyS-lineHeight, 16) * 1px);
```
Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.module.css:21]: we should set relative linge heights

File: src/components/Typography/BodyX/BodyX.test.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+describe('BodyX Component', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  test('renders BodyS correctly', () => {
+    const { asFragment } = render(<BodyS>{'Text'}</BodyS>)
+
+    const text = screen.getByText('Text')
```
Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.test.tsx:34]: Can we add a role like 'paragraph' to be able to retrieve them in tests using roles?

File: src/components/Typography/BodyX/BodyX.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+   */
+  isBold?: boolean,
+}
+
+type BodyXSize = {
+
+  /**
+   * Defines the body font size and line height. Either:
+   *
+   * - s: small - fontSize: 14px, lineHeight: 16px by default;
```
Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.tsx:87]: let's use relative line heights

File: src/components/Typography/HX/HX.tsx [hunk 8]
```diff
@@ -49,9 +49,8 @@ export type HXProps = {
   copyable?: boolean | CopyConfig,
 
   /**
-   * Displays ellipsis when text overflows. By default, the text will be truncated if it
-   * exceeds the length of 2 rows, showing all informational content in a tooltip.
-   * The attribute is further customizable according to the following data model:
+   * Displays ellipsis when text overflows.
```
Comment (Giogia) [inline src/components/Typography/HX/HX.tsx:52]: As we discussed, I would add specific roles for heading components (h1, h2, ...)

**Review by Giogia — CHANGES_REQUESTED** [review]

File: src/components/Typography/BodyX/BodyX.module.css (hunk 5 above)
Comment (RiccardoCorona-MiaPlatform) [reply src/components/Typography/BodyX/BodyX.module.css:21]: Apparently not 🤷🏻

File: src/components/Typography/BodyX/BodyX.tsx (hunk 7 above)
Comment (RiccardoCorona-MiaPlatform) [reply src/components/Typography/BodyX/BodyX.tsx]: The `lineHeight` measurements already look correct 🤷🏻

File: src/components/Typography/BodyX/BodyX.module.css (hunk 5 above)
Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.module.css:21]: default is 24 now :wink:

Comment (Giogia) [inline src/components/Typography/BodyX/BodyX.module.css:21]:
```suggestion
  line-height: calc(var(--typography-bodyS-lineHeight, 24) * 1px);
```

File: src/components/Typography/HX/HX.stories.tsx [hunk 9]
```diff
@@ -40,7 +53,7 @@ export default meta
 type Story = StoryObj<typeof meta>
 
 export const HXComponent: Story = {
-  args: { ...meta.args, level: 1 },
+  args: { ...meta.args, level: 1, role: 'h1' },
```
Comment (Giogia) [inline src/components/Typography/HX/HX.stories.tsx:56]:
```suggestion
  args: { ...meta.args, level: 1 },
```

is it necessary?

**Review by Giogia — APPROVED** [review]

**Review by Danielecina — APPROVED** [review]

**Pull request merged**